extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java

```diff
@@ -654,22 +648,22 @@ private static void extractClientVersionAndKeyFromHtmlSearchResultsPage()
 
     @Nullable
     private static String getClientVersionFromServiceTrackingParam(
-            @Nonnull final Stream<JsonObject> serviceTrackingParamsStream,
+            @Nonnull final JsonArray serviceTrackingParams,
             @Nonnull final String serviceName,
             @Nonnull final String clientVersionKey) {
-        return serviceTrackingParamsStream.filter(serviceTrackingParam ->
-                        serviceTrackingParam.getString("service", "")
-                                .equals(serviceName))
-                .flatMap(serviceTrackingParam -> serviceTrackingParam.getArray("params")
-                        .stream())
-                .filter(JsonObject.class::isInstance)
-                .map(JsonObject.class::cast)
-                .filter(param -> param.getString("key", "")
-                        .equals(clientVersionKey))
-                .map(param -> param.getString("value"))
-                .filter(paramValue -> !isNullOrEmpty(paramValue))
-                .findFirst()
-                .orElse(null);
+        for (int i = 0 ; i < serviceTrackingParams.size(); i++) {
+            JsonObject item = serviceTrackingParams.getObject(i);
+            if (item != null && item.getString("service").equals(serviceName) && item.getArray("params") != null) {
+                JsonArray paramsArray = item.getArray("params");
+                for (int j = 0 ; j < paramsArray.size(); j++) {
+                    JsonObject paramItem = paramsArray.getObject(i);
+                    if (paramItem != null && paramItem.getString("key").equals(clientVersionKey)) {
+                        return paramItem.getString("value");
+                    }
+                }
+            }
+        }
+        return null;
```

Comment:
Mmmh, I think the problem was that `serviceTrackingParamsStream` above was being reused (which is [not possible](https://stackoverflow.com/questions/36255007/is-there-any-way-to-reuse-a-stream)), hence the strange exception. I don't think you needed to switch from the `Stream` API to a for loop. Please revert this change and just add this at the top of this function's call chain:
```java
                .filter(JsonObject.class::isInstance)
                .map(JsonObject.class::cast)
```
Then test if this actually works (to make sure the issue is not something else)